pkg/blobstore/local/file_backed_location_record_array.go

```diff
@@ -0,0 +1,88 @@
+package local
+
+import (
+	"encoding/binary"
+	"sync"
+
+	"github.com/buildbarn/bb-storage/pkg/filesystem"
+)
+
+func serializeLocationRecord(locationRecord LocationRecord) []byte {
+	serialised := make([]byte, 60)
+	for i := 0; i < 32; i++ {
+		serialised[i] = locationRecord.Key.Digest[i]
+	}
+	binary.LittleEndian.PutUint32(serialised[32:36], locationRecord.Key.Attempt)
+	binary.LittleEndian.PutUint64(serialised[36:44], uint64(locationRecord.Location.BlockID))
+	binary.LittleEndian.PutUint64(serialised[44:52], uint64(locationRecord.Location.OffsetBytes))
+	binary.LittleEndian.PutUint64(serialised[52:60], uint64(locationRecord.Location.SizeBytes))
+
+	return serialised
+}
+
+func deserializeLocationRecord(record []byte) LocationRecord {
+	var keyDigest [32]byte
+	copy(keyDigest[0:32], record[0:32])
+	keyAttempt := binary.LittleEndian.Uint32(record[32:36])
+	locationBlockID := int(binary.LittleEndian.Uint64(record[36:44]))
+	locationOffset := int64(binary.LittleEndian.Uint64(record[44:52]))
+	locationSize := int64(binary.LittleEndian.Uint64(record[52:60]))
+
+	key := LocationRecordKey{
+		Digest:  keyDigest,
+		Attempt: keyAttempt,
+	}
+
+	location := Location{
+		BlockID:     locationBlockID,
+		OffsetBytes: locationOffset,
+		SizeBytes:   locationSize,
+	}
+
+	return LocationRecord{
+		Key:      key,
+		Location: location,
+	}
+}
+
+type fileBackedLocationRecordArray struct {
+	recordFile filesystem.FileReadWriter
+	lock       sync.Mutex
+}
+
+// NewFileBackedLocationRecordArray creates a persistent LocationRecordArray.
+// Works by using a file as an array-like structure, writing serialised
+// LocationRecords next to each other and calculating where to Read/Write using
+// the length of a serialised record multiplied by the index.
+func NewFileBackedLocationRecordArray(recordFile filesystem.FileReadWriter) LocationRecordArray {
+	return &fileBackedLocationRecordArray{
+		recordFile: recordFile,
+		lock:       sync.Mutex{},
+	}
+}
+
+func (lra *fileBackedLocationRecordArray) Get(index int) (LocationRecord, error) {
+	lra.lock.Lock()
+	defer lra.lock.Unlock()
+
+	offset := int64(index * 60)
+	record := make([]byte, 60)
+	_, err := lra.recordFile.ReadAt(record, offset)
+	if err != nil {
```

Comment:
I would write this as:
```go
var record [60]byte
if _, err := lra.recordFile.ReadAt(record[:], int64(index) * 60); err != nil {
        return LocationRecord{}, err
}
```
- You can use arrays here instead of slices.
- `int64(index * 60)` is more likely to suffer from integer overflows than `int64(index) * 60`. A plain `int` may be 32 bits in size.
- Use scoped variables where possible.

The same holds for the code in `Put()`.